packages/plugin-ext/src/common/arrays.ts

```diff
@@ -44,3 +44,10 @@ export interface Splice<T> {
     readonly deleteCount: number;
     readonly toInsert: T[];
 }
+
+/**
+ * @returns True if the input object is readonlyarray.
+ */
+export function isArrayAsReadonly(arg: unknown): arg is readonly unknown[] {
+    return Array.isArray(arg);
+}
```

Comment:
We can simplify to:

```suggestion
/**
 * @returns `true` if the `arg` is a `ReadonlyArray`.
 */
export function isReadonlyArray(arg: unknown): arg is readonly unknown[] {
    // Since TypeScript does not properly narrow down typings for `ReadonlyArray` we need to help it.
    return isArray(arg);
}
```